### PR TITLE
Index for documentation and tables of content

### PR DESCRIPTION
**Pull request by muddv**
Added index.md which lists every file in docs/ folder with links to every chapter
Also added table of contents for each individual doc

### CODE REVIEW

File: docs/writing-a-plugin.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Writing a PostCSS Plugin
 
+* [Writing a PostCSS Plugin](#writing-a-postcss-plugin)
```
Comment (ai) [inline docs/writing-a-plugin.md:3]: Do we really need top links to the root?

Comment (muddv) [reply docs/writing-a-plugin.md:3]: Not really, I can delete them, added for consistency, but it's probably an overkill

Comment (ai) [reply docs/writing-a-plugin.md:3]: Let’s delete them.

Everything else looks good.

File: docs/guidelines/runner.md [hunk 2]
```diff
@@ -14,6 +14,25 @@ See also [ClojureWerkz’s recommendations] for open source projects.
 [`gulp‑postcss`]:      https://github.com/w0rm/gulp-postcss
 [`postcss-cli`]:       https://github.com/postcss/postcss-cli
 
+**Table of Contents**
+
+* [PostCSS Runner Guidelines](#postcss-runner-guidelines)
```
Comment (ai) [inline docs/guidelines/runner.md:19]: We need to remove all other roots too

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
+**Index**
```
Comment (ai) [inline docs/index.md:1]:
```suggestion
# Documentation

```

File: docs/writing-a-plugin.md (hunk 1 above)
Comment (ai) [inline docs/writing-a-plugin.md:3]:
```suggestion
**Table of Contents**

* [Writing a PostCSS Plugin](#writing-a-postcss-plugin)
```

Comment (muddv) [reply docs/writing-a-plugin.md]: deleted

Comment (muddv) [reply docs/writing-a-plugin.md]: changed this

File: docs/writing-a-plugin.md [hunk 4]
```diff
@@ -1,3 +1,13 @@
+
+**Table of Contents**
```
Comment (ai) [inline docs/writing-a-plugin.md:2]:
```suggestion
**Table of Contents**

```

File: docs/architecture.md [hunk 5]
```diff
@@ -4,8 +4,6 @@ General overview of the PostCSS architecture.
 It can be useful for everyone who wishes to contribute to the core or develop a better understanding of the tool.
 
 **Table of Contents**
-
-* [PostCSS Architecture](#postcss-architecture)
   * [Overview](#overview)
```
Comment (ai) [inline docs/architecture.md:7]: Can I also ask you to remove the indent of the block (for consistency) and we are ready to merge

Comment (muddv) [reply docs/architecture.md:7]: This is also a root link, but I did not add it, should I remove it too?

Comment (ai) [reply docs/architecture.md:7]: Yes, for consistancy

Comment (muddv) [reply docs/architecture.md:7]: done!

**Comment by ai** [comment]
Great. Now our docs looks better. Thanks.

**Pull request merged**